src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -913,8 +913,12 @@ private async Task LoadPropertyOrFieldAsync(Newtonsoft.Json.Serialization.JsonPr
                     throw new InvalidOperationException("The JSON property '" + propertyName + "' is defined multiple times on type '" + parentType.FullName + "'.");
 
                 var requiredAttribute = propertyAttributes.TryGetIfAssignableTo("System.ComponentModel.DataAnnotations.RequiredAttribute");
+                
+                var propertyRequired = 
+                    objectContract?.ItemRequired != null && 
+                    !propertyAttributes.OfType<JsonPropertyAttribute>().Any() ? objectContract.ItemRequired : property.Required;
```

Comment:
Add `.Any(a => a._required != null)` to check whether `Required` is "really" overridden, see
https://github.com/JamesNK/Newtonsoft.Json/blob/d48558b53b0a516bfa15c1af50231ea1ba7c2454/Src/Newtonsoft.Json/JsonPropertyAttribute.cs#L46